src/components/common/forms/SignInForm.tsx

```diff
@@ -1,31 +1,63 @@
 import React from "react";
 import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+
 import { Button, Input } from "@components/common";
+import { signInRequest } from "@/services";
+import { urlFormat, setLocalStorage } from "@/utils";
+import { Pages } from "@/pages";
+
 import styles from "./Forms.module.css";
 
 const SignInForm = () => {
-  const { register, handleSubmit } = useForm();
-  const onSubmit = (data: object) => {
-    //logic of submission
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setError,
+    clearErrors,
+  } = useForm();
+  const navigate = useNavigate();
+
+  const onSubmit = async (data: object) => {
+    const response = await signInRequest({
+      user: data,
+    });
+    if (typeof response === "object") {
+      setLocalStorage("user", response);
+      navigate(urlFormat(Pages.Main));
+    } else {
+      setError("serverError", { type: "custom", message: response });
+    }
   };
 
+  const isFormValid = Object.keys(errors).length === 0;
+
   return (
     <form
       className={`${styles.form} flex-column`}
       onSubmit={handleSubmit(onSubmit)}>
+      <span className={isFormValid ? "" : styles.error}>
+        {isFormValid
+          ? ""
+          : errors?.serverError?.message?.toString() ||
+            "Both fields are required"}
+      </span>
       <Input
         label="email"
         type="email"
-        stylesName="signIn"
+        stylesName={isFormValid ? "signIn" : "error"}
         register={register}
         required
+        onChange={() => clearErrors()}
       />
       <Input
         label="password"
         type="password"
-        stylesName="signIn"
+        stylesName={isFormValid ? "signIn" : "error"}
         register={register}
         required
+        onChange={() => clearErrors()}
```

Comment:
Same question

Reply:
Here I also want to clear just the custom error `"serverError"`, I have updated it too!